Automattic/jetpack
# Search: Add setting that triggers running the instant search auto config.

Adds a way to trigger the auto config of search so that we can have a smooth default onboarding experience.

I've only done minimal testing so far, but it did work. I'm also not sure this is the best way to go. I especially didn't retest some of the corner cases: https://github.com/Automattic/jetpack/blob/master/modules/search/class-jetpack-instant-search.php#L391

Testing:
- start with a site on a free plan, a sidebar theme (e.g. 2016), and with no jp search widgets configured in the sidebar
- run D41051 on your sandbox and sandbox the public api
- run dev calypso (unless https://github.com/Automattic/wp-calypso/pull/40545 has landed)
- Purchase a search plan
- Your site should have the search widgets get added to both the main sidebar and the overlay sidebar

Fast CL testing: 
- see D41051
